src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = null) {
+    this.name = name;
+    this.health = health || 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = null) {
+    super(name, health);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = null) {
+    super(name, health);
+    Animal.alive.push(this);
```

Comment:
The line `Animal.alive.push(this);` is redundant in the `Carnivore` constructor because it is already called in the `Animal` constructor. This results in the `Carnivore` instance being added twice to the `Animal.alive` array.